src/basic/virt.c

```diff
@@ -867,10 +867,23 @@ int running_in_userns(void) {
 int running_in_chroot(void) {
         int r;
 
+        /* If we're PID1, /proc may not be mounted (and most likely we're not in a chroot). But PID1 will
+         * mount /proc, so all other programs can assume that if /proc is *not* available, we're in some
+         * chroot. */
```

Comment:
actually, if we are PID 1 then the comparison of `/proc/1/root` with `/` doesn't really make much sense, it's of course going to be true, we compare our root with our root, wich is definitely true ;-)

Hence, maybe add a check early on:

```c
if (getpid_cached() == 1)
        return false;
```

and then drop the getpid_cached() below

Reply:
Updated.